src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -779,6 +779,31 @@ def projection_in_str(self) -> str:
             )
         return self._projection_in_str
 
+    def with_subqueries(self, subquery_plans: List[SnowflakePlan]) -> "SelectStatement":
+        """Update pre-actions, post-actions and schema to capture necessary subquery_plans
+        encountered during plan resolution."""
+        for plan in subquery_plans:
+            for query in plan.queries[:-1]:
+                if self.pre_actions is None:
+                    self.pre_actions = []  # pragma: no cover
```

Comment:
How can we have no cover for this line, most of the case , pre_actions should be None, right?